File: java/sourceafis/src/sourceafis/templates/SerializedFormat.java
package sourceafis.templates;

import java.io.IOException;
import java.io.InputStream;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.OutputStream;


public final class SerializedFormat extends TemplateFormatBase<Template>{
    @Override
	public Template exportTemplate(TemplateBuilder builder)
    {
        return new Template(builder);
    }

    @Override
    public TemplateBuilder importTemplate(Template template)
    {
        return template.toTemplateBuilder();
    }
    @Override
    public void serialize(OutputStream stream, Template template)
    {
       
		try {
			ObjectOutputStream formatter = new ObjectOutputStream(stream);
			formatter.writeObject(template);
		} catch (IOException e) {
	    	throw new RuntimeException(e);
		}
     }
    @Override
    public Template deserialize(InputStream stream)
    {
    	try {
          ObjectInputStream formatter=new ObjectInputStream(stream);
       	  return (Template)formatter.readObject();
		} catch (ClassNotFoundException e) {
		  throw new RuntimeException(e);
		}catch(IOException e){
          throw new RuntimeException(e);
    	}
    }
}

File: SourceAFIS/Templates/SerializedFormat.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using System.Runtime.Serialization;
using System.Runtime.Serialization.Formatters.Binary;
using SourceAFIS.General;

namespace SourceAFIS.Templates
{
    public sealed class SerializedFormat : TemplateFormatBase<Template>
    {
        public override Template Export(TemplateBuilder builder)
        {
            return new Template(builder);
        }

        public override TemplateBuilder Import(Template template)
        {
            return template.ToTemplateBuilder();
        }

        public override void Serialize(Stream stream, Template template)
        {
            BinaryFormatter formatter = new BinaryFormatter();
            formatter.Serialize(stream, template);
        }

        public override Template Deserialize(Stream stream)
        {
            BinaryFormatter formatter = new BinaryFormatter();
            return formatter.Deserialize(stream) as Template;
        }
    }
}
